### PR TITLE
Fix for hard-coded nu-get Newtonsoft usage in C# SDK causing Unity IL2CPP builds to crash

### PR DESCRIPTION
**Pull request by jakub-grzesiowski**
fix: fix hard-coded nu-get Newtonsoft usage in C# SDK causing Unity IL2CPP builds to crash

Fixes issue of Unity SDK not being able to have a fully independent IJsonPluggableLibrary implementation because of direct nu-get Newtonsoft usage in C# SDK (Unity has it's own version that works with IL2CPP builds)

### CODE REVIEW

**Comment by jakub-grzesiowski** [comment]
**IMPORTANT**: Probably best to only review and not merge before 06.04.2024 when I come back since this needs a Unity counterpart anyway

File: src/Api/PubnubApi/EndPoint/PubSub/SubscribeManager2.cs [hunk 1]
```diff
@@ -114,7 +114,7 @@ public SubscribeManager2(PNConfiguration pubnubConfig, IJsonPluggableLibrary jso
             if (!string.IsNullOrEmpty(responseTuple.Item1) && responseTuple.Item2 == null)
             {
                 PNStatus status = new PNStatus(null, PNOperationType.PNSubscribeOperation, PNStatusCategory.PNConnectedCategory, channels, channelGroups);
-                HandshakeResponse handshakeResponse = JsonConvert.DeserializeObject<HandshakeResponse>(responseTuple.Item1);
+                HandshakeResponse handshakeResponse = jsonLibrary.DeserializeToObject<HandshakeResponse>(responseTuple.Item1);
```
Comment (mohitpubnub) [inline src/Api/PubnubApi/EndPoint/PubSub/SubscribeManager2.cs:117]: 👍🏻

File: src/Api/PubnubApi/EventEngine/Subscribe/Effects/EmitMessagesHandler.cs [hunk 2]
```diff
@@ -14,18 +12,18 @@ public class EmitMessagesHandler : EffectHandler<EmitMessagesInvocation>
 	{
 		private readonly Dictionary<string, Type> channelTypeMap;
 		private readonly Dictionary<string, Type> channelGroupTypeMap;
-		private readonly JsonSerializer serializer;
+		private readonly IJsonPluggableLibrary jsonPluggableLibrary;
```
Comment (mohitpubnub) [inline src/Api/PubnubApi/EventEngine/Subscribe/Effects/EmitMessagesHandler.cs:15]: 👍🏻

**Review by mohitpubnub — APPROVED** [review]
LGTM!

**Comment by jakub-grzesiowski** [comment]
@pubnub-release-bot release as v6.20.1.0

**Pull request merged**